Wrappers/Python/ccpi/viewer/CILViewerBase.py

```diff
@@ -0,0 +1,278 @@
+import vtk
+from ccpi.viewer.utils.io import SaveRenderToPNG
+
+SLICE_ORIENTATION_XY = 2  # Z
+SLICE_ORIENTATION_XZ = 1  # Y
+SLICE_ORIENTATION_YZ = 0  # X
+
+CONTROL_KEY = 8
+SHIFT_KEY = 4
+ALT_KEY = -128
+
+SLICE_ACTOR = 'slice_actor'
+OVERLAY_ACTOR = 'overlay_actor'
+HISTOGRAM_ACTOR = 'histogram_actor'
+HELP_ACTOR = 'help_actor'
+CURSOR_ACTOR = 'cursor_actor'
+CROSSHAIR_ACTOR = 'crosshair_actor'
+LINEPLOT_ACTOR = 'lineplot_actor'
+WIPE_ACTOR = 'wipe_actor'
+
+
+class ViewerEventManager(object):
+
+    def __init__(self):
+        # If all values are false it signifies no event
+        self.events = {
+            "PICK_EVENT": False,  # left  mouse
+            "WINDOW_LEVEL_EVENT": False,  # alt + right mouse + move
+            "ZOOM_EVENT": False,  # shift + right mouse + move
+            "PAN_EVENT": False,  # ctrl + right mouse + move
+            "CREATE_ROI_EVENT": False,  # ctrl + left mouse
+            "DELETE_ROI_EVENT": False,  # alt + left mouse
+            "SHOW_LINE_PROFILE_EVENT": False,  # l
+            "UPDATE_WINDOW_LEVEL_UNDER_CURSOR": False,  # Mouse move + w
+            "RECTILINEAR_WIPE": False  # activated by key 2, updates by mouse move
+        }
+
+    def __str__(self):
+        return str(self.events)
+
+    def On(self, event):
+        self.events[event] = True
+
+    def Off(self, event):
+        self.events[event] = False
+
+    def setAllInactive(self):
+        self.events = {x: False for x in self.events}
+
+    def isActive(self, event):
+        return self.events[event]
+
+    def isAllInactive(self):
+        """Returns True if all events are inactive"""
+        return all(not x for x in self.events.values())
+
+
+class CILViewerBase():
+
```

Comment:
needs to document at least that the interactor needs to be told which render window it's meant to use. See comment above.

Reply:
made a setInteractorStyle method in the base class